validate.go

```diff
@@ -182,39 +182,39 @@ func checkProcess(process rspec.Process, rootfs string) {
 }
 
 //Linux only
-func checkLinux(spec rspec.Linux, hostname string, rootfs string) {
+func checkLinux(spec rspec.Spec) {
 	utsExists := false
 
-	if len(spec.UIDMappings) > 5 {
+	if len(spec.Linux.UIDMappings) > 5 {
 		logrus.Fatalf("Only 5 UID mappings are allowed (linux kernel restriction).")
 	}
-	if len(spec.GIDMappings) > 5 {
+	if len(spec.Linux.GIDMappings) > 5 {
 		logrus.Fatalf("Only 5 GID mappings are allowed (linux kernel restriction).")
 	}
 
-	for index := 0; index < len(spec.Namespaces); index++ {
-		if !namespaceValid(spec.Namespaces[index]) {
-			logrus.Fatalf("namespace %v is invalid.", spec.Namespaces[index])
-		} else if spec.Namespaces[index].Type == rspec.UTSNamespace {
+	for index := 0; index < len(spec.Linux.Namespaces); index++ {
+		if !namespaceValid(spec.Linux.Namespaces[index]) {
+			logrus.Fatalf("namespace %v is invalid.", spec.Linux.Namespaces[index])
+		} else if spec.Linux.Namespaces[index].Type == rspec.UTSNamespace {
 			utsExists = true
 		}
 	}
 
-	if !utsExists && hostname != "" {
+	if spec.Platform.OS == "linux" && !utsExists && spec.Hostname != "" {
 		logrus.Fatalf("Hostname requires a new UTS namespace to be specified as well")
```

Comment:
The PR looks good. But I suggest to change this message `Hostname requires a new UTS namespace to be specified as well`. 
How about `On Linux, hostname requires a UTS namespace to be specified as well`?